spring-cloud-dataflow-core/src/main/java/org/springframework/cloud/dataflow/core/StreamDefinition.java

```diff
@@ -153,6 +149,28 @@ public void remove() {
 			throw new UnsupportedOperationException();
 		}
 
+		/**
+		 * Return whether or not this iterator is past the very first module to deploy.
+		 */
+		public boolean hasMoreDownstream() {
+			return i < modules.size() - 2;
```

Comment:
It seems wrong to have a method which is only applicable after `next()` has been called. I think that's a red flag that we aren't expressing the right thing.